internal/rbd/rbd_util.go

```diff
@@ -697,47 +697,58 @@ func (ri *rbdImage) trashRemoveImage(ctx context.Context) error {
 	return nil
 }
 
-func (ri *rbdImage) getCloneDepth(ctx context.Context) (uint, error) {
-	var depth uint
-	vol := rbdVolume{}
+// getCloneDepth walks the parents of the image and returns the number of
+// images in the chain.
+//
+// This function re-uses the ioctx of the image to open all images in the
+// chain. There is no need to open new ioctx's for every image.
+func (ri *rbdImage) getCloneDepth() (uint, error) {
```

Comment:
hey @nixpanic,
Can we add another arguement like `maxDepthToTraverse` which can be used to exit and not traverse further unnecessarily? 

We can pass hard `rbdHardMaxCloneDepth+1` to the function call then.

wdyt?

Reply:
sure, I'll add that

Reply:
Added a 2nd commit to address this.